Assets/Tests/InputSystem/Plugins/UITests.cs

```diff
@@ -3761,6 +3761,264 @@ public IEnumerator UI_WhenCursorIsLockedToScreenCenter_PointerEnterAndExitEvents
         Assert.That(callbackReceiver.events.Any(e => e.type == EventType.PointerExit), Is.True);
     }
 
+    #region Multi Display Tests
+#if UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY
+    [UnityTest]
+    public IEnumerator UI_DisplayIndexMatchesDisplayWithTouchscreenOnScreenSpaceCanvas()
+    {
+        // Setup the Test Scene
+        InputSystem.AddDevice<Touchscreen>();
+        var scene = CreateTestUI();
+        var actions = ScriptableObject.CreateInstance<InputActionAsset>();
+        var uiActions = actions.AddActionMap("UI");
+        var pointAction = uiActions.AddAction("point", type: InputActionType.PassThrough, binding: "<Touchscreen>/position");
+        var clickAction = uiActions.AddAction("press", type: InputActionType.PassThrough, binding: "<Touchscreen>/press");
+        actions.Enable();
+        scene.uiModule.point = InputActionReference.Create(pointAction);
+        scene.uiModule.leftClick = InputActionReference.Create(clickAction);
+
+        // Set Display Index to 1
+        byte targetDisplay = 1;
+        scene.camera.targetDisplay = targetDisplay; // Only need to set camera since this is scene space canvas
+        yield return null;
+
+        // Touch Display 1
+        BeginTouch(1, new Vector2(100, 100), queueEventOnly: true, displayIndex: targetDisplay);
+        yield return null;
+        EndTouch(1, new Vector2(100, 100), queueEventOnly: true, displayIndex: targetDisplay);
+        yield return null;
+
+        // Verify Display Index Was Set
+        Assert.AreEqual(2, scene.parentReceiver.events.Count);
+        Assert.AreEqual(targetDisplay, scene.parentReceiver.events[0].pointerData.displayIndex);
+        Assert.AreEqual(targetDisplay, scene.parentReceiver.events[1].pointerData.displayIndex);
+
+        // Set Display Index to 0
+        scene.camera.targetDisplay = 0;
+        yield return null;
+
+        // Touch Display 1
+        scene.ClearEvents();
+        BeginTouch(1, new Vector2(100, 100), queueEventOnly: true, displayIndex: targetDisplay);
+        yield return null;
+        EndTouch(1, new Vector2(100, 100), queueEventOnly: true, displayIndex: targetDisplay);
+        yield return null;
+
+        // Verify Clicks On Display 1 Are Ignored On Display 0
+        Assert.AreEqual(0, scene.parentReceiver.events.Count);
```

Comment:
Do you think it would be worth adding tests that verify old behavior as well?

Reply:
@3iD0loN I'm not sure I understand.  Do you mean writing a test to make sure valid clicks are _not_ discarded?  That should be covered by these tests.

But I think I may not be understanding.

Reply:
I spoke with Joel offline. He meant to add a test that verifies displayIndex is not set when UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY isn't set.

I'm not sure this is possible to test in any meaningful way because pointerData.displayIndex didn't exist before this review.

Reply:
As I understand this UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY will be defined by Unity. Hence if the Unity version the tests are running against define this, a certain behavior is expected, but if this is not defined another ("legacy") behavior is expected. If the test in question verify a behavior previously not covered by existing tests, it may be valuable that the test exercises behaviors of when UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY is defined as well as when it is not defined since Input System package may be combined with any Unity version e.g. 2023.x 2022.x, 2021.x 2020.x.
Since UGUI changes comes with trunk at least there is no mismatch combination between UGUI and Input System package in addition to this as I understand.

Reply:
The first clause of your first comment is correct.  There is no new functionality when UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY isn't set so we should be good.

In reference to your second comment: 
If I'm understanding correctly, it sounds like you agree with Joel: that there should be a test added for when UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY is not defined.  Is that correct?

If so, I'm not sure what that test would be testing. This PR adds a new field called "displayIndex" the compilation of which is gated by this flag UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY.

When UNITY_INPUT_SYSTEM_ENABLE_MULTIDISPLAY is not defined, there is no new displayIndex to test.  The most you could do is use reflection to verify that there is no property called displayIndex, but that to me doesn't seem like a meaningful test.

I could TOTALLY be misunderstanding here so please correct me if I've got it wrong.

Reply:
Good call on adding the parentReceiver.event.position check.  I just added that to the PR.

I think I understand what you're saying. I'll add a test that clicks on display 2 in version 2019 and make sure the click works.